src/yt-music/YTSearchHandler.ts

```diff
@@ -0,0 +1,142 @@
+import { Page } from "puppeteer";
+import { SearchHandler, SearchListSubscriber, Song } from "../user-controls/SearchHandler";
+import crypto from 'crypto';
+
+export class YTSearchHandler implements SearchHandler {
+
+  private page: Page;
+  public songList: Song[];
+  private subscribers: {};
+
+  constructor (page: Page) {
+    this.page = page;
+    this.songList = [];
+    this.subscribers = {};
+  }
+  
+  subscribe (callback: SearchListSubscriber) {
+    const subscriberId = crypto.randomBytes(8).toString('hex');
+    this.subscribers[subscriberId] = callback;
+    return subscriberId;
+  }
+
+  unsubscribe (subscriberID: string) {
+    delete this.subscribers[subscriberID];
+  }
+
+  async play (playID: string) {
+    if (playID === 'No results found.') return;
+    // Have to do it this way becuase some play buttons may be off the screen or under elements that will break things if clicked.
+    await this.page.evaluate(
+      (playID) => {
+        // @ts-ignore because it will have click method.
+        document.querySelector(`.${playID}`).click();
```

Comment:
I think you can explicitly designate playID as HTMLElement using $eval instead of evaluate. So something like:
```
await this.page.$eval(`.${playID}`, (el: HTMLElement) => el.click());
```
Not 100% sure, but something like that could work and avoid the ts-ignore

Reply:
Yup, haven't refactored at all yet so you'll find little gold nuggets like this lying around lol